lib/os_detect.dart

```diff
@@ -24,37 +32,59 @@ String get operatingSystemVersion => OperatingSystem.current.version;
 ///
 /// This value is `false` if the operating system is a specialized
 /// version of Linux that identifies itself by a different name,
-/// for example Android (see [isAndroid]).
+/// for example Android (see [isAndroid]),
+/// or if the code is running inside a browser (see [isBrowser]).
+@pragma('vm:prefer-inline')
```

Comment:
Notice that this annotation will not make guarantees, the new `@pragma('vm:platform-const')` will make guarantees (/cc @sstrickl )

Even if the method is inlined, it may happen in the C++ compiler backend. That has also a tree-shaker, but we want to ensure it's tree shaken before our kernel-level global analysis pass. We have this new platform-const annotation for that purpose.

If there's subsets of expression/statement language missing to make it work, @sstrickl can implement that.

Reply:
I'm not expecting this getter to be platform-const, that's going to be impossible when we also have the ability to do runtime overrides.

However, if inlined twice, and all but one `OSKind` subclass has been tree-shaken, the compiler can still deduce that the value is always true or false. (And it works! So don't get worse at treeshaking!)